src/a_star_search/core.clj

```diff
@@ -14,19 +14,19 @@
 (defn -main
   "Search for a hard-coded N-puzzle target state."
   [& args]
-  (let [start-state (np/->State [[0 1 3] [4 2 5] [7 8 6]] [0 0])
-        goal-state (np/->State [[0 1 2] [3 4 5] [6 7 8]] [0 0])
+  (let [start-state (tri/->State [[1] [1 1] [1 1 1] [1 1 1 1] [1 1 1 1 0]])
+        goal-state (tri/->State [[0] [0 0] [1 0 0] [0 0 0 0] [0 0 0 0 0]])
         max-states 1000000
         costs nil
-        came-from (alg/breadth-first-search np/children max-states start-state goal-state)
-        ; [came-from costs] (alg/shortest-path np/children (constantly 1)
-        ;                                      max-states start-state goal-state )
-        ; [came-from costs] (alg/shortest-path np/children np/prefer-horizontal-cost
+        ;came-from (alg/breath-first-search tri/children max-states start-state goal-state)
```

Comment:
You somehow introduced a mis-spelling (`breadth`, not `breath`) here, so it you uncomment this line it doesn't work.